client/rest-high-level/src/test/java/org/elasticsearch/client/SearchIT.java

```diff
@@ -0,0 +1,226 @@
+        StringEntity doc4 = new StringEntity("{\"field\":\"value4\", \"num\":100}", ContentType.APPLICATION_JSON);
+        client().performRequest("PUT", "/index/type/4", Collections.emptyMap(), doc4);
+        StringEntity doc5 = new StringEntity("{\"field\":\"value5\", \"num\":100}", ContentType.APPLICATION_JSON);
+        client().performRequest("PUT", "/index/type/5", Collections.emptyMap(), doc5);
+        client().performRequest("POST", "/index/_refresh");
+    }
+
+    public void testSearchNoQuery() throws IOException {
+        SearchRequest searchRequest = new SearchRequest();
+        SearchResponse searchResponse = execute(searchRequest, highLevelClient()::search, highLevelClient()::searchAsync);
+        assertNull(searchResponse.getAggregations());
+        assertNull(searchResponse.getSuggest());
+        assertEquals(Collections.emptyMap(), searchResponse.getProfileResults());
+        assertThat(searchResponse.getTook().nanos(), greaterThan(0L));
+        assertEquals(0, searchResponse.getFailedShards());
+        assertEquals(5, searchResponse.getSuccessfulShards());
+        assertEquals(5, searchResponse.getTotalShards());
+        assertEquals(0, searchResponse.getShardFailures().length);
+        assertEquals(5, searchResponse.getHits().totalHits);
+        assertEquals(5, searchResponse.getHits().getHits().length);
+        for (SearchHit searchHit : searchResponse.getHits().getHits()) {
+            assertEquals("index", searchHit.getIndex());
+            assertEquals("type", searchHit.getType());
+            assertThat(Integer.valueOf(searchHit.getId()), both(greaterThan(0)).and(lessThan(6)));
+            assertEquals(1.0f, searchHit.getScore(), 0);
+            assertEquals(-1L, searchHit.getVersion());
+            assertNotNull(searchHit.getSourceAsMap());
+            assertEquals(2, searchHit.getSourceAsMap().size());
+            assertTrue(searchHit.getSourceAsMap().containsKey("field"));
+        }
+    }
+
+    public void testSearchMatchQuery() throws IOException {
+        SearchRequest searchRequest = new SearchRequest();
+        searchRequest.source(new SearchSourceBuilder().query(new MatchQueryBuilder("field", "value1")));
+        SearchResponse searchResponse = execute(searchRequest, highLevelClient()::search, highLevelClient()::searchAsync);
+        assertNull(searchResponse.getAggregations());
+        assertNull(searchResponse.getSuggest());
+        assertEquals(Collections.emptyMap(), searchResponse.getProfileResults());
+        assertThat(searchResponse.getTook().nanos(), greaterThan(0L));
+        assertEquals(0, searchResponse.getFailedShards());
+        assertEquals(5, searchResponse.getSuccessfulShards());
+        assertEquals(5, searchResponse.getTotalShards());
+        assertEquals(0, searchResponse.getShardFailures().length);
+        assertEquals(1, searchResponse.getHits().totalHits);
+        assertEquals(1, searchResponse.getHits().getHits().length);
```

Comment:
Some if these assertions look quiet similar for most test cases here, maybe they can be moved to a single helper method?